include/ccf/crypto/cose_signer.h

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ccf/crypto/pem.h"
+
+namespace ccf::crypto
+{
+  class COSESigner
+  {
+  public:
+    virtual bool sign(
+      std::span<uint8_t> payload,
```

Comment:
```suggestion
      std::span<const uint8_t> payload,
```

I think I'd even `const std::span<const uint8_t>& payload` to make it clear `sign()` doesn't update the span itself either, but I can see that this is debatable.

Reply:
I don't think `const std::span<T>&` makes sense, as well as `const std::span<T>`.

Passing `std::span` is as good, + can be more efficient due to enabling passing via registers, and has same `const`ness guarantees over the content.

I'll use `const size_t`, however, this does indeed make sense.